diff --git a/deps/cairo/build.rs b/deps/cairo/build.rs
index c070f351d..a214ea989 100644
--- a/deps/cairo/build.rs
+++ b/deps/cairo/build.rs
@@ -210,6 +210,6 @@ fn cairo() {
 }
 
 fn main() {
-    pixman();
-    cairo();
+    println!("cargo:rustc-link-lib=pixman-1");
+    println!("cargo:rustc-link-lib=cairo")
 }
diff --git a/deps/freetype/build.rs b/deps/freetype/build.rs
index b4a8407b8..c3b16d422 100644
--- a/deps/freetype/build.rs
+++ b/deps/freetype/build.rs
@@ -235,9 +235,9 @@ fn git_submodule_update() {
 }
 
 fn main() {
-    zlib();
-    libpng();
-    freetype();
+    println!("cargo:rustc-link-lib=z");
+    println!("cargo:rustc-link-lib=png");
+    println!("cargo:rustc-link-lib=freetype");
     let out_dir = env::var("OUT_DIR").unwrap();
     println!("cargo:outdir={}", out_dir);
     println!("cargo:rustc-env=MACOSX_DEPLOYMENT_TARGET=10.12");
diff --git a/deps/harfbuzz/build.rs b/deps/harfbuzz/build.rs
index 98c372e71..37cd13001 100644
--- a/deps/harfbuzz/build.rs
+++ b/deps/harfbuzz/build.rs
@@ -65,7 +65,7 @@ fn git_submodule_update() {
 }
 
 fn main() {
-    harfbuzz();
+    println!("cargo:rustc-link-lib=harfbuzz");
     let out_dir = env::var("OUT_DIR").unwrap();
     println!("cargo:outdir={}", out_dir);
     println!("cargo:rustc-env=MACOSX_DEPLOYMENT_TARGET=10.12");
diff --git a/window/Cargo.toml b/window/Cargo.toml
index 3cfa26dc9..96e3dd0d1 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -270,7 +270,7 @@ windows = "0.33.0"
 winreg = "0.10"
 x11 = "2.21"
 xcb = "1.3"
-xcb-imdkit = { version="0.3", git="https://github.com/wezterm/xcb-imdkit-rs.git", rev="212330f7c6c37794d78061a773e8f4f9e4785bbb"}
+xcb-imdkit = { version="0.3", git="https://github.com/wezterm/xcb-imdkit-rs.git", rev="212330f7c6c37794d78061a773e8f4f9e4785bbb", features = ["use-system-lib"]}
 xkbcommon = "0.7.0"
 xml-rs = "0.8"
 zbus = "4.2"
